src/Microsoft.DotNet.Build.CloudTestTasks/AzureHelper.cs

```diff
@@ -0,0 +1,373 @@
+            string method, 
+            DateTime now, 
+            HttpWebRequest request, 
+            string ifMatch = "", 
+            string md5 = "", 
+            string size = "", 
+            string contentType = "")
+        {
+            var stringToSign = string.Format(
+                "{0}\n\n\n{1}\n{5}\n{6}\n\n\n{2}\n\n\n\n{3}{4}", 
+                method, 
+                (size == string.Empty) ? string.Empty : size, 
+                ifMatch, 
+                GetCanonicalizedHeaders(request), 
+                GetCanonicalizedResource(request.RequestUri, storageAccount), 
+                md5, 
+                contentType);
+
+            var signatureBytes = Encoding.UTF8.GetBytes(stringToSign);
+            var hmacsha256 = new HMACSHA256(Convert.FromBase64String(storageKey));
+            var authorizationHeader = "SharedKey " + storageAccount + ":"
+                                      + Convert.ToBase64String(hmacsha256.ComputeHash(signatureBytes));
+            return authorizationHeader;
+        }
+
+        /// <summary>
+        /// The create container sas token.
+        /// </summary>
+        /// <param name="accountName">
+        /// The account name.
+        /// </param>
+        /// <param name="containerName">
+        /// The container name.
+        /// </param>
+        /// <param name="key">
+        /// The key.
+        /// </param>
+        /// <param name="accessType">
+        /// The access type.
+        /// </param>
+        /// <param name="validityTimeInDays">
+        /// The validity time in days.
+        /// </param>
+        /// <returns>
+        /// The <see cref="string"/>.
+        /// </returns>
+        public static string CreateContainerSasToken(
+            string accountName, 
+            string containerName, 
+            string key, 
+            SasAccessType accessType, 
+            int validityTimeInDays)
+        {
+            var signedpermissions = string.Empty;
+            if (accessType == SasAccessType.read)
+            {
+                signedpermissions += "r";
+            }
+
+            if (accessType == SasAccessType.write)
+            {
+                signedpermissions += "wdl";
+            }
+
+            var signedstart = DateTime.UtcNow.ToString("O");
+            var signedexpiry = DateTime.UtcNow.AddDays(validityTimeInDays).ToString("O");
+            var canonicalizedresource = "/blob/" + accountName + "/" + containerName;
+            var signedidentifier = string.Empty;
+            var signedversion = StorageApiVersion;
+
+            var stringToSign = signedpermissions + "\n" + signedstart + "\n" + signedexpiry + "\n"
+                               + canonicalizedresource + "\n" + signedidentifier + "\n" + string.Empty + "\n" + string.Empty + "\n"
+                               + signedversion + "\n" + string.Empty + "\n" + string.Empty + "\n" + string.Empty + "\n" + string.Empty + "\n" + string.Empty;
```

Comment:
Adding string.Empty is weird.  If you're using them to signify a placeholder, that'd make a lot more sense if the structure you were building was documented immediately beforehand.